From b9e3841445914e43ecf42975f2d8a8f3c8239e25 Mon Sep 17 00:00:00 2001
From: Hang Xiaoqian <hangxiaoqian@wxiat.com>
Date: Wed, 25 May 2022 10:54:25 +0800
Subject: [PATCH] sw64: change the value of physical_id in /proc/cpuinfo
Patch-mainline: Not yet, from openEuler
References: bsn#22
openEuler-commit: b9e3841445914e43ecf42975f2d8a8f3c8239e25
Modified-by-SEL: No


Sunway inclusion
category: feature
bugzilla: https://gitee.com/openeuler/kernel/issues/I56OLG

--------------------------------

The value of physical_id in /proc/cpuinfo should be
the socket_id of the processor.

Signed-off-by: Hang Xiaoqian <hangxiaoqian@wxiat.com>

Signed-off-by: Gu Zitao <guzitao@wxiat.com>
Signed-off-by: Guoqing Jiang <guoqing.jiang@suse.com>
---
 arch/sw_64/kernel/setup.c | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/arch/sw_64/kernel/setup.c b/arch/sw_64/kernel/setup.c
index faae7cb3c5a1..4d39db5fb1ef 100644
--- a/arch/sw_64/kernel/setup.c
+++ b/arch/sw_64/kernel/setup.c
@@ -907,7 +907,7 @@ show_cpuinfo(struct seq_file *f, void *slot)
 				"physical id\t: %d\n"
 				"bogomips\t: %lu.%02lu\n",
 				cpu_freq, cpu_data[i].tcache.size >> 10,
-				cpu_to_rcid(i),
+				cpu_topology[i].package_id,
 				loops_per_jiffy / (500000/HZ),
 				(loops_per_jiffy / (5000/HZ)) % 100);
 
-- 
2.33.0

